packages/host/tests/integration/components/ai-assistant-panel/codeblocks-test.gts

```diff
@@ -279,4 +279,291 @@ module('Integration | ai-assistant-panel | codeblocks', function (hooks) {
 
     await percySnapshot(assert);
   });
+
+  test('handles nested pre tags', async function (assert) {
+    let roomId = await renderAiAssistantPanel(`${testRealmURL}Person/fadhlan`);
+    let messageWithNestedPreTags = `I'll help you create an example with pre tags in HTML. Here's a useful example that demonstrates different ways to use the pre tag:
+\`\`\`html
+<!-- Basic pre tag example -->
+<pre>
+This is preformatted text
+    It preserves both spaces
+        and line breaks
+exactly as written
+</pre>
+
+<!-- Pre tag with code -->
+<pre>
+<code>
+function sayHello() {
+    console.log("Hello World!");
+}
+</code>
+</pre>
+
+<!-- Pre tag with styling -->
+<pre style="background-color: #f4f4f4; padding: 15px; border-radius: 5px;">
+const data = {
+    name: "John",
+    age: 30,
+    city: "New York"
+};
+</pre>
+
+<!-- Pre tag with HTML entities -->
+<pre>
+&lt;html&gt;
+    &lt;head&gt;
+        &lt;title&gt;Sample Page&lt;/title&gt;
+    &lt;/head&gt;
+    &lt;body&gt;
+        &lt;h1&gt;Hello World!&lt;/h1&gt;
+    &lt;/body&gt;
+&lt;/html&gt;
+</pre>
+\`\`\`
+
+\`\`\`typescript
+  <<<<<<< SEARCH
+    let a = 1;
+    let c = 3;
+  =======
+    let a = 2;
+\`\`\`
+
+These examples show different ways to use the \`<pre>\` tag:
+1. Basic preformatted text
+2. Code display with \`<pre>\` and \`<code>\` combined
+3. Styled \`<pre>\` block
+4. HTML code display using entities
+
+You can use these in your HTML documents to display formatted text, code snippets, or any content where you want to preserve spacing and line breaks exactly as written.`;
+
+    simulateRemoteMessage(
+      roomId,
+      '@aibot:localhost',
+      {
+        body: messageWithNestedPreTags,
+        formatted_body: messageWithNestedPreTags,
+        msgtype: 'org.text',
+        format: 'org.matrix.custom.html',
+        isStreamingFinished: true,
+      },
+      {
+        origin_server_ts: new Date(2024, 0, 3, 12, 30).getTime(),
+      },
+    );
+
+    await waitFor('[data-test-message-idx="0"]');
+
+    assert
+      .dom('.monaco-editor')
+      .exists({ count: 2 }, 'Should have 2 monaco editors');
+
+    let monacoContent = getMonacoContent();
+    assert.ok(
```

Comment:
I think a better test would be to just assert the whole content instead of many `.includes()`